# AlwaysValid

- `AlwaysValid()`

Validates any input as valid.

```python
v.alwaysValid().validate('whatever')  # true
```

## Categorization

- Booleans

## Changelog

Version | Description
--------|-------------
  1.0.0 | Created

***
See also:

- [AlwaysInvalid](AlwaysInvalid.md)