# AlwaysInvalid

- `AlwaysInvalid()`

Validates any input as invalid.

```python
v.alwaysInvalid().validate('whatever')  # false
```

## Categorization

- Booleans

## Changelog

Version | Description
--------|-------------
  1.0.0 | Created

***
See also:

- [AlwaysValid](AlwaysValid.md)
- [When](When.md)